### PR TITLE
Support weighted target groups in forward listener rules

### LINKED ISSUE #10942: Support ALB weighted target groups
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Application Load Balancers now support weighted target groups routing.

### New or Affected Resource(s)

* [`aws_lb_listener`](https://www.terraform.io/docs/providers/aws/r/lb_listener.html)
* [`aws_lb_listener_rule`](https://www.terraform.io/docs/providers/aws/r/lb_listener_rule.html)

### Potential Terraform Configuration

```hcl
resource "aws_lb_listener_rule" "example" {
  action {
    type = "forward"
    ...
  }
}
```

### References

[Announcement](https://aws.amazon.com/about-aws/whats-new/2019/11/application-load-balancer-simplifies-deployments-support-for-weighted-target-groups/).
[Blog post](https://aws.amazon.com/blogs/aws/new-application-load-balancer-simplifies-deployment-with-weighted-target-groups/).

Related:
* https://github.com/terraform-providers/terraform-provider-aws/issues/8126

Requires:
* https://github.com/terraform-providers/terraform-provider-aws/pull/10940

### PR DESCRIPTION
**Pull request by goodspark**
**Opening this as draft to get feedback on my initial approach.** Tested this and was able to create a weighted forward target group rule on an existing load balancer listener.

TODOS
- [ ] Tests
- [ ] Add to default_actions in aws_lb_listener_rule as well
- [ ] Doc updates
- [ ] Add acceptance tests

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #10942 

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
* resource/aws_lb_listener_rule: Add support for multiple, weighted target groups in forward rules.
* resource/aws_lb_listener: Add support for multiple, weighted target groups in default actions.
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

File: aws/resource_aws_lb_listener_rule.go [hunk 1]
```diff
@@ -74,6 +74,56 @@ func resourceAwsLbbListenerRule() *schema.Resource {
 							DiffSuppressFunc: suppressIfActionTypeNot("forward"),
 						},
 
+						"forward": {
```
Comment (goodspark) [inline aws/resource_aws_lb_listener_rule.go]: Here's what the schema would look like in a TF file:

```hcl
resource "aws_alb_listener_rule" "inf" {
  listener_arn = "arn:aws:elasticloadbalancing:...."
  condition {
    path_pattern {
      values = ["/blahblah/*"]
    }
  }
  action {
    type = "forward"
    forward {
      target_group {
        arn = "arn:aws:elasticloadbalancing:...."
        weight = 2
      }

      target_group {
        arn = "arn:aws:elasticloadbalancing:...."
        weight = 1
      }
    }
  }
}
```

**Comment by rdelcampog** [comment]
Looks good! I really need this functionality merged asap for my project. :)

**Comment by djtecha** [comment]
Would this be added to the aws_lb_listener resource too? It would be very helpful since we can't define 0 actions in that resource currently.

**Comment by goodspark** [comment]
That's on the list of TODOs. I'd like some feedback from the maintainers that this is how they want this implemented and if I'm missing any edge cases before I spend more time on it.

**Comment by rdelcampog** [comment]
@bflad Can you please take a look to this PR? I think this is a very interesting feature announced a few months ago that needs to be in the provider asap :)

Thanks.

**Comment by damdo** [comment]
Any thought on this one @bflad? I'd be very valuable.
Thank you 🙌

**Comment by rdelcampog** [comment]
@goodspark I think you should implement the TODOs and maybe @bflad can review your changes in a next phase.

**Comment by rdelcampog** [comment]
> @goodspark I think you should implement the TODOs and maybe @bflad can review your changes in a next phase.

Hi, I've been trying to implement the remaining things in your PR. I'm not a maintainer so I opened a new PR #12574. Hope you don't mind.

**Comment by tinuz74** [comment]
When will this be released?

**Comment by breathingdust** [comment]
This was released in v2.65.0 of the provider 🎉 !

**Pull request closed without merging**